### DISCUSSION TITLE
difference between angular and radial feature in the descriptor

### DISCUSSION THREAD

**Discussion by echen361** [deepmodeling/deepmd-kit#2067] [Q&A]
I understand that knowing the radial distance between atoms is important, but I don't see how a two atom descriptor can effectively use angle(full information) as input. 

The DeepMD-kit paper explains that the angle of the atoms is determined when the coordinates of the atoms is fixed, but couldn't the entire coordinate system be rotated, which makes the angle irrelevant? For example, how can 2 atoms describe bonds bending and dihedral angle forces? 

Even the MEAM potential and the newly updated VASP 6 machine learning uses 3 atoms if angle is an input.

Therefore I don't really see how using angles as an input for 2 atom descriptor can provide anymore information. Any insights?

**Comment by wanghan-iapcm** [comment]
There is no "2 atom" descriptor in deepmd. 

The input information is an atom and the coordinates of **all** its near neighbors, so the deepmd is a muti-body potential energy. 

You may be confused with "2-body embedding" which referring to the descriptor that only use distance in the embedding net. Note that even the "2-body embedding" descriptor is multi-body and considers all the angles between near neighbors.

**Reply by echen361** [reply]
I see now. Thanks for the clarification!

**Comment by omidshy** [comment]
As it's stated [here](https://docs.deepmodeling.com/projects/deepmd/en/v2.1.5/model/train-se-e2-a.html) the "se_e2_a" descriptor is "constructed from all information (both angular and radial) of atomic configurations". However, I can not find any info regarding the angular information in the DeepPot-SE [paper](https://arxiv.org/abs/1805.09003). 

It's still not clear to me how the angular information can be captured if only the pair-wise distances of the central atom to all its near neighbors are used to construct the descriptors?

**Reply by njzjz** [reply]
For `se_e2_a`, the distance is a vector, but not a scalar.

**Reply by wanghan-iapcm** [reply]
> As it's stated [here](https://docs.deepmodeling.com/projects/deepmd/en/v2.1.5/model/train-se-e2-a.html) the "se_e2_a" descriptor is "constructed from all information (both angular and radial) of atomic configurations". However, I can not find any info regarding the angular information in the DeepPot-SE [paper](https://arxiv.org/abs/1805.09003).
> 
> It's still not clear to me how the angular information can be captured if only the pair-wise distances of the central atom to all its near neighbors are used to construct the descriptors?

The environment matrix encodes the full information (including the angular) of the neighbors.
